src/benchmarks/micro/libraries/System.Collections/Sort.cs

```diff
@@ -59,6 +62,20 @@ public class Sort<T> where T : IComparable<T>
         [Benchmark]
         public void List() => _lists[_iterationIndex++].Sort();
 
+        [GlobalSetup(Target = nameof(ImmutableArray))]
+        public void SetupImmutableArray() =>
+            _immutableArray = Immutable.ImmutableArray.CreateRange(GenerateValues());
+
+        [Benchmark]
+        public ImmutableArray<T> ImmutableArray() => _immutableArray.Sort();
+
+        [GlobalSetup(Target = nameof(ImmutableList))]
+        public void SetupImmutableList() =>
+            _immutableList = Immutable.ImmutableList.CreateRange(GenerateValues());
+
+        [Benchmark]
+        public ImmutableList<T> ImmutableList() => _immutableList.Sort();
```

Comment:
Most of this benchmark assumes mutable sort, so it's setup involves populating N lists/arrays upfront and then sorting each of them on subsequent iterations. That doesn't make sense for immutable collections, but will the fixed invocation count throw off the measurement at all here?